hpx/lcos/local/detail/sliding_semaphore.hpp

```diff
@@ -35,6 +35,13 @@ namespace hpx { namespace lcos { namespace local { namespace detail
           : max_difference_(max_difference), lower_limit_(lower_limit), cond_()
         {}
 
+        void set_max_difference(std::int64_t max_difference,
+                                std::int64_t lower_limit)
+        {
+            max_difference_ = max_difference;
+            lower_limit_    = lower_limit;
+        }
```

Comment:
We will have to acquire a lock to set those values in any case. Otherwise, platforms with a weaker memory model might end up writing half of the values in racey situations.